taxonomy/management/commands/refresh_course_skills.py

```diff
@@ -61,11 +70,39 @@ def handle(self, *args, **options):
         """
         Entry point for management command execution.
         """
-        if options['course'] == [] and options['args_from_database'] is False:
-            parser = self.create_parser('manage.py', 'refresh_course_skills')
-            parser.error('Either course or args_from_database argument must be present')
+        if not (options['args_from_database'] or options['all'] or options['course']):
+            raise InvalidCommandOptionsError('Either course, args_from_database or all argument must be provided.')
 
         if options['args_from_database']:
             options = self.get_args_from_database()
 
-        utils.refresh_course_skills(options)
+        LOGGER.info('[TAXONOMY] Refresh Course Skills. Options: [%s]', options)
+
+        if options['all']:
+            courses = get_course_metadata_provider().get_all_courses()
+        elif options['course']:
+            courses = get_course_metadata_provider().get_courses(course_ids=options['course'])
+            if not courses:
+                raise CourseMetadataNotFoundError(
+                    'No course metadata was found for following courses. {}'.format(options['course'])
+                )
+        else:
+            raise InvalidCommandOptionsError('Either course or all argument must be provided.')
+
+        LOGGER.info('[TAXONOMY] Refresh course skills process started.')
+        success_courses, skipped_courses, failures = utils.refresh_course_skills(courses, options['commit'])
```

Comment:
are we doing update in chunks? I am unable to find.

Reply:
@muhammad-ammar chunking is on the course discovery side. Course Discovery is providing data. You will see that code in upcoming PR.

Reply:
Yes. this method is returning the generator.